ext/standard/string.c

```diff
@@ -4530,6 +4531,11 @@ PHP_FUNCTION(parse_str)
 		sapi_module.treat_data(PARSE_STRING, res, &ret);
 		ZVAL_COPY_VALUE(arrayArg, &ret);
 	}
+
+	/* parse_str() tries to parse as much as possible (per treat_data behavior), hence we just ignore possible HashCollisionErrors */
+	if (EG(exception) && EG(exception)->ce == zend_ce_hash_collision_error) {
+		zend_clear_exception();
+	}
```

Comment:
When max_input_vars is exceeded, only a warning is emitted and no exceptions either.

Perhaps we should make this function emit a warning in this case too?